src/components/AuthPage/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { Component } from 'react'
+import axios from 'axios';
+
+import paths from '../../paths.config';
+
+interface IAuthPageProps {
+  location: { hash: string },
+  history: { push: (url: string) => void }
+}
+
+export default class AuthPage extends Component<IAuthPageProps, {}> {
+  constructor(props: IAuthPageProps) {
+    super(props);
+    this.state = {
+      fetching: true,
+    }
+  }
+  async componentDidMount() {
+    const token = this.props.location.hash.split("access_token=")[1].split("&")[0]
```

Comment:
looks very nice, but... If you try to load /auth page after logging in with google you will have `Unhandled Rejection (TypeError): Cannot read property 'split' of undefined` error :)

Reply:
I agree. This is a very naive implementation. Do you have any better solutions?